packages/harpguru-core/src/hooks/use-menu-animation-values/use-menu-animation-values.ts

```diff
@@ -66,56 +64,43 @@ export const useMenuAnimationValues = (
 
   const { shortEdge, longEdge } = getWindowDimensions()
   const animationDuration = 300
-  const stashMenuTiming = useTimingTransition(isMenuStashed, {
-    duration: animationDuration,
-    easing: Easing.inOut(Easing.ease),
-  })
-  const hideLabelTiming = useTimingTransition(isLabelHidden, {
-    duration: animationDuration,
-    easing: Easing.inOut(Easing.ease),
+  const stashMenuTiming = useDerivedValue(() => {
+    return withTiming((isMenuStashed ? 1 : 0), {
+      duration: animationDuration,
+      easing: Easing.inOut(Easing.ease),
+    })
   })
 
-  const scaledFullX = multiply(longEdge, menuScaleTranslationFactor)
-  const stashedX = sub(longEdge, scaledFullX)
-  const stashXVector = multiply(stashedX, stashXDirection)
-  const stashXValue = interpolate(stashMenuTiming, {
-    inputRange: [0, 1],
-    outputRange: [0, stashXVector],
-  })
-  const stashedY = multiply(shortEdge, menuStashedYOffsetFactor)
-  const scaledStashedY = multiply(stashedY, menuScaleTranslationFactor)
-  const stashedYVector = multiply(scaledStashedY, stashYDirection)
-  const stashYValue = interpolate(stashMenuTiming, {
-    inputRange: [0, 1],
-    outputRange: [0, stashedYVector],
-  })
-  const hideLabelVector = multiply(
-    useScaledMenuLabelProtrusion(),
-    stashXDirection
-  )
-  const hideXValue = interpolate(hideLabelTiming, {
-    inputRange: [0, 1],
-    outputRange: [0, hideLabelVector],
-  })
-  const slideX = add(stashXValue, hideXValue)
-  const slideY = stashYValue
-  const scale = interpolate(stashMenuTiming, {
-    inputRange: [0, 1],
-    outputRange: [1, menuStashedScale],
-  })
-  const backgroundColor = interpolateColor(stashMenuTiming, {
-    inputRange: [0, 1],
-    outputRange: [colors.pageColor, colors.inertOutline],
-  })
-  const opacity = interpolate(stashMenuTiming, {
-    inputRange: [0, 1],
-    outputRange: [overlayOpacity, 1],
+  const hideLabelTiming = useDerivedValue(() => {
+    return withTiming((isLabelHidden ? 1 : 0), {
+      duration: animationDuration,
+      easing: Easing.inOut(Easing.ease),
+    })
   })
+  const hideLabelVector = useScaledMenuLabelProtrusion() * stashXDirection
+
+  const slideX = useDerivedValue(() => {
+    const scaledFullX = longEdge * menuScaleTranslationFactor
+    const stashedX = longEdge - scaledFullX
+    const stashXVector = stashedX * stashXDirection
+    const stashXValue = interpolate(stashMenuTiming.value, [0, 1], [0, stashXVector])
+    const hideXValue = interpolate(hideLabelTiming.value, [0, 1], [0, hideLabelVector])
+    return stashXValue + hideXValue
+  }) // }, TODO: add dependencies? )
```

Comment:
The simplest approach is to not include dependencies at all. That way the worklet will be reloaded every time any of it's dependents change. This might be over cautious in some situations, but I'm not here to optimise unnecessarily. We just need to get this upgrade out.

Reply:
Evidence: https://docs.swmansion.com/react-native-reanimated/docs/api/hooks/useDerivedValue/

Reply:
All of these have been simplified now.